### PR TITLE
conditionals: fix handling of empty string values

### LINKED ISSUE #10168: Modify filter conditions does not work when filter is used as input processor
## Bug Report

**Describe the bug**
The `condition` key seems to overlap with `condition` parameter for `modify` filter. This causes configuration errors when using `modify` filter as an input processor.  I tried replacing `modify` filter with `content_modifier` processor but it does not support conditions such as `key_exists`. I have also noticed that when `value` field value is an empty string it causes `content_modifier` processor errors.

**To Reproduce**
Following input processor causes error messages during `fluentbit` init:
```
pipeline:
  inputs:
    - name: tail
      tag: ecs.*
      path: /var/lib/docker/containers/*/*-json.log
      processors:
        logs:
          - name: modify
            condition: key_does_not_exist container
            set: container unknown
```

This example results in following error logs:
```
Fluent Bit v4.0.0
* Copyright (C) 2015-2025 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

______ _                  _    ______ _ _             ___  _____
|  ___| |                | |   | ___ (_) |           /   ||  _  |
| |_  | |_   _  ___ _ __ | |_  | |_/ /_| |_  __   __/ /| || |/' |
|  _| | | | | |/ _ \ '_ \| __| | ___ \ | __| \ \ / / /_| ||  /| |
| |   | | |_| |  __/ | | | |_  | |_/ / | |_   \ V /\___  |\ |_/ /
\_|   |_|\__,_|\___|_| |_|\__| \____/|_|\__|   \_/     |_(_)___/


[2025/04/02 20:49:15] [error] [processor] condition must be a map
[2025/04/02 20:49:15] [error] failed to set condition for processor 'modify'
[2025/04/02 20:49:15] [error] failed to load 'logs' processors
```

I have tried replacing `modify` filter with `content_modifier` processor with following config:
```
pipeline:
  inputs:
    - name: tail
      tag: ecs.*
      path: /var/lib/docker/containers/*/*-json.log
      processors:
        logs:
          - name: content_modifier
            condition:
              op: and
              rules:
                - field: container
                  op: not_regex // not sure how else to check if field exists
                  value: ".+"
            key: container
            action: insert
            value: unknown
```
Unfortunately, last example does not add `container` key when it does not exist.

**Expected behavior**
Be able to use filter conditions when using `modify` filter as input processor.
Be able to add keys when key/field does not exists already.

**Your Environment**

* Version used: 4.0.0
* Configuration: YAML
* Environment name and version (e.g. Kubernetes? What version?): AWS ECS
* Operating System and version: Linux
* Filters and plugins: Modify

### PR DESCRIPTION
**Pull request by niedbalski**
## Summary
- Allow empty string values for equality comparisons in conditionals
- Add proper handling of string conditions in processor
- Fix handling condition types in processor units
- Add test cases for string condition handling and empty values

Fixes #10168

### CODE REVIEW

**Comment by leonardo-albertovich** [comment]
Please don't remove review requests in the future.

File: src/flb_conditionals.c [hunk 1]
```diff
@@ -66,8 +66,14 @@ static struct flb_condition_rule *rule_create(const char *field,
     switch (op) {
     case FLB_RULE_OP_EQ:
     case FLB_RULE_OP_NEQ:
+        /* Allow empty string values for equality comparisons */
+        if (!value) {
```
Comment (leonardo-albertovich) [inline src/flb_conditionals.c:70]: Please modify this to explicitly compare with `NULL`.

File: src/flb_conditionals.c [hunk 2]
```diff
@@ -66,8 +66,14 @@ static struct flb_condition_rule *rule_create(const char *field,
...
+            return NULL;
+        }
+        break;
     case FLB_RULE_OP_REGEX:
     case FLB_RULE_OP_NOT_REGEX:
+        /* Regex patterns must not be empty */
         if (!value || !((char *)value)[0]) {
```
Comment (leonardo-albertovich) [inline src/flb_conditionals.c:77]: Please modify this to explicitly compare with `NULL`.
Please refactor the second part of the term to make it clearer.

File: src/flb_conditionals.c (hunk 1 above)
Comment (leonardo-albertovich) [reply src/flb_conditionals.c:70]: Also, wouldn't this be covered by the conditional in line 62? (which should be modified to explicitly compare the pointer values with `NULL`)

File: tests/internal/processor_conditional.c [hunk 3]
```diff
@@ -51,21 +51,35 @@ static void cleanup_test_resources(struct flb_config *config, struct flb_process
...
 }
 
+/* Create a string variant for testing non-kvlist condition handling */
+static struct cfl_variant *create_string_condition_variant(const char *value)
+{
+    struct cfl_variant *variant;
+
+    /* Using cfl_variant_create_from_string instead of a non-existent cfl_variant_set_string */
+    variant = cfl_variant_create_from_string((char *)value);
+    if (!variant) {
```
Comment (leonardo-albertovich) [inline tests/internal/processor_conditional.c:76]: If you are not doing anything (ie. reporting) if `cfl_variant_create_from_string` returns `NULL` why do you have this code? returning `variant` would be the same.

**Review by leonardo-albertovich — CHANGES_REQUESTED** [review]
Please change all the pointer validations to explicitly compare against `NULL` (not just the ones I marked).

**Comment by railsharipov** [comment]
Thank you for working on this. I hope this will be merged soon.
